rust/protocol/src/sealed_sender.rs

```diff
@@ -877,10 +1015,219 @@ mod sealed_sender_v2 {
             }
         }
 
-        HKDF::new(3)?.derive_secrets(&agreement_key_input, LABEL_DH_S, AUTH_TAG_LEN)
+        let boxed_result =
+            HKDF::new(3)?.derive_secrets(&agreement_key_input, LABEL_DH_S, AUTH_TAG_LEN)?;
+        boxed_result
+            .as_ref()
+            .try_into()
+            .map_err(|_| SignalProtocolError::BadKeyLength(curve::KeyType::Djb, boxed_result.len()))
```

Comment:
Same thing about not being a Curve25519 key, but also I think it's okay to `expect()` this one because we just generated it from HKDF; that is, it's not data passed in from outside. (Really I need to change HKDF's interface to generate into an output slice instead of returning a box, and then all these conversions go away. I have a half-written patch for that already so you don't need to do it.)

Reply:
Done! Changes are in cf048dd.

> I have a half-written patch for that already so you don't need to do it.

Yay!! I remember that part was a lot of work when I tried to tackle it :)